### PR TITLE
feat: add modules for getting Talos support bundle

### PR DESCRIPTION
**Pull request by Unix4ever**
Slightly modified version of Talos support bundle getter code.

### CODE REVIEW

**Review by frezbo — DISMISSED** [review]

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,78 @@
+module github.com/siderolabs/go-talos-support
+
+go 1.22.1
+
+require (
+	github.com/containerd/containerd v1.7.13
+	github.com/cosi-project/runtime v0.4.0-alpha.9
+	github.com/dustin/go-humanize v1.0.1
+	github.com/hashicorp/go-multierror v1.1.1
+	github.com/siderolabs/talos/pkg/machinery v1.6.0-alpha.2.0.20240307132915-3c8f51d707b8
```
Comment (smira) [inline go.mod]: this looks like a strange tag?

Comment (Unix4ever) [reply go.mod]: That's `github.com/siderolabs/talos/pkg/machinery main` then `go mod tidy`. This lib needs the latest main as it uses extracted modules

Comment (smira) [reply go.mod]: we have 1.7.0-alpha.1 tagged at least, so it can't be this tag

Comment (Unix4ever) [reply go.mod]: I wonder why it doesn't pick it up automatically 🤔

Comment (Unix4ever) [reply go.mod]: I see only alpha.0 btw, which doesn't have my changes.

Comment (smira) [reply go.mod]: ok, still should be 1.7 :)

**Review by smira — APPROVED** [review]

**Pull request merged**